File: 백준/Bronze/24416. 알고리즘 수업 － 피보나치 수 1/알고리즘 수업 － 피보나치 수 1.java
import java.io.*;

public class Main{
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        
        int n = Integer.parseInt(br.readLine().trim());

        int[] fibCount = new int[n];
        fibCount[0] = 1;
        fibCount[1] = 1;

        for (int i = 2; i < n; i++) {
            fibCount[i] = fibCount[i-1] +fibCount[i-2];
        }

        bw.write(String.valueOf(fibCount[n-1]) + " " + String.valueOf(n-2) + '\n');
		bw.flush();
		bw.close();
		br.close();
	}
}

File: 백준/Bronze/24416. 알고리즘 수업 － 피보나치 수 1/알고리즘 수업 － 피보나치 수 1.py
import sys

input = sys.stdin.readline

n = int(input().strip())

fib_count = list(range(n))
fib_count[0], fib_count[1] = 1, 1

for i in range(2, n):
    fib_count[i] = fib_count[i-1] + fib_count[i-2]

sys.stdout.write(f"{fib_count[n-1]} {n-2}\n")